### PR TITLE
Add address share screen

### PR DESCRIPTION
**Pull request by aminsato**

### CODE REVIEW

File: app/src/main/java/com/vultisig/wallet/presenter/qr_address/QrAddressScreen.kt [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+import androidx.navigation.NavHostController
+import com.vultisig.wallet.R
+import com.vultisig.wallet.presenter.common.rememberQRBitmapPainter
+import com.vultisig.wallet.ui.components.QrBorder
+import com.vultisig.wallet.ui.theme.Theme
+import com.vultisig.wallet.ui.theme.dimens
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun QrAddressScreen(navController: NavHostController) {
```
Comment (YGolybev) [inline app/src/main/java/com/vultisig/wallet/presenter/qr_address/QrAddressScreen.kt:43]: Please, let's keep everything internal where possible; it'll help us to modularize the application later

File: app/src/main/java/com/vultisig/wallet/ui/navigation/Navigation.kt [hunk 2]
```diff
@@ -128,5 +128,11 @@ internal sealed class Destination(
     data object VultisigToken : Destination(route = "settings/vultisig_token")
     data object LanguageSetting : Destination(route = "settings/language")
     data object CurrencyUnitSetting : Destination(route = "settings/currency")
+    class QrAddressScreen(val address:String) : Destination(route = "vault_details/qr_address_screen/$address"){
```
Comment (YGolybev) [inline app/src/main/java/com/vultisig/wallet/ui/navigation/Navigation.kt:131]:
```suggestion
    data class QrAddressScreen(val address:String) : Destination(route = "vault_details/qr_address_screen/$address"){
```

File: app/src/main/java/com/vultisig/wallet/presenter/qr_address/QrAddressScreen.kt [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+            val screenWidth = LocalConfiguration.current.screenWidthDp
+            val address = viewmodel.address!!
+            val qrBoxSize = ((screenWidth * .8).coerceAtMost(300.0)).dp
+            val imageAndBorderSpace = 30.dp
+            val imageSizeInDp = qrBoxSize - 2 * imageAndBorderSpace
+            val whiteSpace = 10.dp
+
+            Text(
+                text = address,
+                style = Theme.menlo.titleMedium,
```
Comment (YGolybev) [inline app/src/main/java/com/vultisig/wallet/presenter/qr_address/QrAddressScreen.kt:99]: In Figma, and it seems that on iOS it's menlo.body1
```suggestion
                style = Theme.menlo.body1,
```

File: app/src/main/java/com/vultisig/wallet/presenter/qr_address/QrAddressScreen.kt [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+                modifier = Modifier
+                    .align(Alignment.TopCenter)
+                    .padding(top = 128.dp)
+            )
+
+            Box(
+                Modifier
+                    .clip(shape = RoundedCornerShape(qrBoxSize.div(10)))
```
Comment (YGolybev) [inline app/src/main/java/com/vultisig/wallet/presenter/qr_address/QrAddressScreen.kt:107]: It's probably just fixed 10.dp size of corners

**Review by YGolybev — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/vultisig/wallet/ui/components/QrBorder.kt [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+                relativeLineTo(halfNegativeSegment + halfBorderWidth, zero)
+                relativeMoveTo(negSegment, zero)
+                relativeLineTo(negSegment, zero)
+                relativeMoveTo(negSegment, zero)
+                relativeLineTo(halfNegativeSegment, zero)
+                arcTo(
+                    Rect(
+                        topLeft = Offset(halfBorderWidth, 4 * segment - halfBorderWidth),
+                        bottomRight = Offset(
+                            segment + halfBorderWidth, 5 * segment - halfBorderWidth
```
[the remark this answers is not in the document]
Comment (aminsato) [reply app/src/main/java/com/vultisig/wallet/ui/components/QrBorder.kt:72]: I aimed to replicate the design from Figma to ensure consistency and fidelity in the output. 
Here is the current output
Screenshot 2024-05-17 at 12 51 12 PM

However, if we opt to use Stroke with PathEffect, we would have
Screenshot 2024-05-17 at 12 51 39 PM

File: app/src/main/java/com/vultisig/wallet/presenter/qr_address/QrAddressScreen.kt (hunk 4 above)
Comment (YGolybev) [reply app/src/main/java/com/vultisig/wallet/presenter/qr_address/QrAddressScreen.kt:107]: And why not to fix size of the QR border at 10.dp too?

File: app/src/main/java/com/vultisig/wallet/ui/components/QrBorder.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (YGolybev) [reply app/src/main/java/com/vultisig/wallet/ui/components/QrBorder.kt:72]: Isn't that what [intervals](https://developer.android.com/reference/kotlin/androidx/compose/ui/graphics/PathEffect#dashPathEffect(kotlin.FloatArray,kotlin.Float)) are for?

Or is it impossible to use those, to replicate design in Figma?

[the remark this answers is not in the document]
Comment (aminsato) [reply app/src/main/java/com/vultisig/wallet/ui/components/QrBorder.kt:72]: ok, then we would have sth like this
telegram-cloud-photo-size-4-5925036968746533315-y
We can not always make sure that corners are same as Figma unless we do some calculation on line width and barcode size and etc. 

I open to use Stroke with PathEffect  and intervals, let me know if it make works for you

[the remark this answers is not in the document]
Comment (johnnyluo) [reply app/src/main/java/com/vultisig/wallet/ui/components/QrBorder.kt:72]: I think this is fine

[the remark this answers is not in the document]
Comment (YGolybev) [reply app/src/main/java/com/vultisig/wallet/ui/components/QrBorder.kt:72]: Yeah, if PathEffect doesn't work, current solution is good

File: app/src/main/java/com/vultisig/wallet/presenter/qr_address/QrAddressScreen.kt (hunk 1 above)
Comment (aminsato) [reply app/src/main/java/com/vultisig/wallet/presenter/qr_address/QrAddressScreen.kt]: Done

**Review by YGolybev — APPROVED** [review]

**Pull request merged**